src/gui_impl/commands/open_file_gui_command.cpp

```diff
@@ -100,6 +103,15 @@ void OpenFileGuiCommand::evaluateEnabledness ()
 //-------------------------------------------------------------------------
 void OpenFileGuiCommand::open ()
 {
+    //by YL
+    //close the previous file before opening a new one
+    //thus using less memory
+    CloseFileGuiCommand closeObject;
+
+    closeObject.closeFile();
+
```

Comment:
I think this should be done later, after a valid file was successfully selected, i.e. right before line 128 (`openFileImpl (file_path)`).